tests/test_compare_dictconfig_vs_dict.py

```diff
@@ -0,0 +1,678 @@
+"""
+This file compares DictConfig methods with the corresponding
+methods of standard python's dict.
+The following methods are compared:
+    __contains__
+    __delitem__
+    __eq__
+    __getitem__
+    __setitem__
+    get
+    pop
+    keys
+    values
+    items
+
+We have separate test classes for the following cases:
+    TestUntypedDictConfig: for DictConfig without a set key_type
+    TestPrimitiveTypeDunderMethods: for DictConfig where key_type is primitive
+    TestEnumTypeDunderMethods: for DictConfig where key_type is Enum
+"""
+from copy import deepcopy
+from enum import Enum
+from typing import Any, Dict
+
+from pytest import fixture, mark, param, raises
+
+from omegaconf import DictConfig, OmegaConf
+from omegaconf.errors import ConfigKeyError, ConfigTypeError, KeyValidationError
+from tests import Enum1
+
+
+@fixture(
+    params=[
+        "a",
+        "b",
+        1,
+        2,
+        123.45,
+        67.89,
+        True,
+        False,
+        Enum1.FOO,
+        Enum1.BAR,
+        "FOO",
+        "BAR",
```

Comment:
I suggest one key per supported key type.
```suggestion
        "str",
        1,
        3.1415,
        True,
        Enum1.FOO,
```